coveo-functools/tests_functools/test_flex_deserializer.py

```diff
@@ -100,10 +103,51 @@ def test_deserialize_thing_or_list_of_things(hint: Type, payload: Any, expected:
         Dict,
         Dict[Any, Any],
         Dict[Union[str, bool], Any],
-        Dict[Any, Union[bytes, MockType]],
-        Dict[Optional[Union[str, int]], Optional[Union[str, MockType, Optional[bool]]]],
+        Dict[str, Optional[Union[bool, int]]],
     ),
 )
 def test_deserialize_dict(hint: Type) -> None:
     """We don't do much with'em, but they gotta work!"""
     assert deserialize(DEFAULT_PAYLOAD, hint=hint) == DEFAULT_PAYLOAD
+
+
+@parametrize(
+    "hint",
+    (
+        Dict[str, MockType],
+        Dict[Any, MockType],
+        Dict[str, Optional[MockType]],
+    ),
+)
+def test_deserialize_dict_with_custom_value_type(hint: Any) -> None:
+    assert deserialize(DEFAULT_MAP_PAYLOAD, hint=Dict[str, MockType]) == DEFAULT_MAP_MOCK

```

Comment:
Is it me, or you don't use the parameter `hint`? Isn't what you want here

```suggestion
    assert deserialize(DEFAULT_MAP_PAYLOAD, hint=hint) == DEFAULT_MAP_MOCK

```
?

Reply:
good catch!